### DISCUSSION TITLE
HASPmota and some widgets - searching for working samples

### DISCUSSION THREAD

**Discussion by waltkind** [arendst/Tasmota#20203] [Support request] [answered]
In an earlier thread I was telling that I switched from openHASP to HASPmota, but I fail to get some widgets to work like they did in openHASP. So I searched the net for working samples but didn't find them.

For example I had a line in openHasp:
`"id":36,"obj":"bar","x":5,"y":240,"w":230,"h":20,"bg_color10":"#6cbe58","bg_color":"#b7dfab","bg_grad_dir10":0,"bg_grad_dir":0,"border_side":0,"pad_top":0,"pad_bottom":0,"pad_left":0,"pad_right":0,"val":65}`
That's not working. So I ommitted some of the attributes with no real success.
In https://tasmota.github.io/docs/HASPmota/#attributes there are no specific attributes for bar.

As workaround I added three lines:
`{"id":36,"obj":"btn","x":5,"y":238,"w":230,"h":20,"bg_color":"green"}
{"id":37,"obj":"btn","x":5,"y":238,"w":0,"h":20,"bg_color":"lime"}
{"id":38,"obj":"label","x":5,"y":240,"w":230,"h":20,"text":"Batterie:","align":"center","text_color":"#000000","text_font":"montserrat-14"}`
The disadvantage is that the bar is now a button and reacts to touch. "click":0 solves this

**Comment by s-hadinger** [comment]
I will try on my side. Let's keep in touch.

**Reply by s-hadinger** [reply]
I fixed the `val` attribute that caused the problem. I'm not sure that all the other attributes are correctly handled. Can you confirm which ones work, and which don't?

**Comment by waltkind** [comment]
I tried again the example with many attributes and got following errors in Berry console:
`HSP: unknown attribute: bg_color10  `
`BRY: failed to run compiled code 'value_error' - Missing erguments, remaining type 'i'`

**Reply by s-hadinger** [reply]
I don't understand, I used the following and have no error:
```
{"page":1,"comment":"---------- Page 1 ----------"}
{"id":0,"bg_color":"#0000A0","text_color":"#FFFFFF"}
{"id":36,"obj":"bar","x":5,"y":140,"w":230,"h":20,"bg_color10":"#6cbe58","bg_color":"#b7dfab","bg_grad_dir10":0,"bg_grad_dir":0,"border_side":0,"pad_top":0,"pad_bottom":0,"pad_left":0,"pad_right":0,"val":65}
```

**Reply by sfromis** [reply]
If you're still having that error, make sure to upgrade to a current Tasmota build, like version 13.3.

**Reply by waltkind** [reply]
Version 13.3 solved the problem of the 'val'-attribute. Thanks.
But the bg_color is not shown. It is only a dark shade. bg_color10 is shown right.

With the new version many of the colors seem to be different. Have there been changes too?

**Reply by waltkind** [reply]
I made some screenshots with no alteration with version 13.2 and 13.3. 
The colors are more faint.
The bg_color of the first bar is 'green' and th second '#FF0000'.
The bg_color10 of all three bars is 'lime'.

![screenshot-1702482536](https://github.com/arendst/Tasmota/assets/83280479/d9b8b47b-5023-4c92-a011-0bdfffd3fe44)
![screenshot-1702552464](https://github.com/arendst/Tasmota/assets/83280479/12d11d73-9e49-4c4c-b4a1-a7f6483fedef)

**Reply by s-hadinger** [reply]
I checked with the LVGL doc. The only meaningful attribute is `bg_color`. You also need to adjust `bg_opa` since it's `51` by default (51/255 = 20% transparency). Setting `bg_opa` to 255 makes the color full scale.

`bg_color10` or `bg_color20` have no effect.

**Reply by waltkind** [reply]
Thank you for that advice. It works now for the bar.
Now I see 2 new effects since the new version:
1. The faint color of the lower buttons results from the attribute "click":0 or "enabled":0.  When I make it clickable again the full color is there. No difference with bg_opa.
2. The switch object does not follow the bg_color10 when engaged. No matter what bg_color or bg_color10 I choose the switch symbol changes to a violet.

**Reply by s-hadinger** [reply, marked as answer]
For 1. it must be a standard behavior of LVGL. You need to dive into LVGL to know if there are other options.

For 2. have you tried `bg_color11`, `bg_color12` and so on ?

**Reply by waltkind** [reply]
Thank you. `bg_color11` works.
I didn't know that it is possible to combine the styling. But that is why I am here to learn.

There must have been a change between V13.2 and V13.3 because as you see in screenshot 1, the faint color was not there. For the bg_color10 of the switch I have no screenshot, but I'm sure that had been OK till the update.

**Reply by s-hadinger** [reply]
Yes indeed. 13.3 tries to align with OpenHASP. See details here:
https://tasmota.github.io/docs/HASPmota/#haspmota-styling

**Comment by waltkind** [comment]
I also loaded your new haspmota.be and put load("haspmota.be") in the first line of autoexec.be, but no change.

**Reply by s-hadinger** [reply]
You don't need to do that, `haspmota.be` is solidified in the firmware.
